lib/dateTime.js

```diff
@@ -0,0 +1,577 @@
+
+	if (isNaN(year) || year < 1753 || year > 9999) {
+		return false;
+	}
+	if (isNaN(month) || month < 1 || month > 12) {
+		return false;
+	}
+	if (isNaN(day) || day < 1 || day > 31) {
+		return false;
+	}
+
+	let allowedDays = 31;
+	if (month === 2) {
+		if ((year % 4 === 0) && ((year % 100 !== 0) || (year % 400 === 0))) {
+			allowedDays = 29;
+		} else {
+			allowedDays = 28;
+		}
+	} else if (month === 4 || month === 6 || month === 9 || month === 11) {
+		allowedDays = 30;
+	}
+
+	if (day > allowedDays) {
+		return false;
+	}
+
+	return true;
+
+}
+
+function prePadByZero(input, maxNum) {
+	input = input.toString();
+	const zero = '0';
+	while (zero.length > 0 && input.length < maxNum) {
+		input = zero + input;
+	}
+	return input;
+}
+
+function processPattern(pattern, replacements) {
+
+	let reStr = '';
+	Object.keys(replacements).forEach(function(key) {
+		reStr += ((reStr === '') ? '' : '|') + key;
+	});
+	const re = new RegExp(reStr, 'g');
+
+	const doReplacements = function(buf) {
+		return buf.replace(re, function(m) {
+			return replacements[m];
+		});
+	};
+
+	let escape = false;
+	let buffer = '';
+	let value = '';
+	for (let i = 0; i < pattern.length; i++) {
+		const c = pattern.charAt(i);
+		if (c === "'") {
+			if (!escape) {
+				value += doReplacements(buffer);
+				buffer = '';
+			}
+			escape = !escape;
+		} else if (escape) {
+			value += c;
+		} else {
+			buffer += c;
+		}
+	}
+	value += doReplacements(buffer);
+
+	return value;
+
+}
+
+export function getDateTimeDescriptor() {
+
+	const language = getLanguage();
+
+	const subtags = language.split('-');
+	const baseLanguage = subtags[0];
+
+	let hour24 = (hour24locales.indexOf(baseLanguage) > -1);
+	if (language === 'zh-tw') {
+		hour24 = false;
+	}
+
+	const timeFormat = getTimeFormat(hour24, language, baseLanguage);
+
+	let dateFormats = ['dddd, MMMM d, yyyy', 'MMM d, yyyy', 'M/d/yyyy', 'MMMM yyyy', 'MMMM d'];
+	let fullTimeFormat = `${timeFormat} ZZZ`;
+	let dayPeriods = ['AM', 'PM'];
+	let months = [
+		['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
+		['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+	];
+	let days = [
+		['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+		['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
+		['S', 'M', 'T', 'W', 'T', 'F', 'S']
+	];
+	let firstDayOfWeek = (mondayFirstDayLocales.indexOf(baseLanguage) > -1) ? 1 : 0;
+	let weekendStartDay = 6;
+	let weekendEndDay = 0;
+
+	switch (baseLanguage) {
+		case 'ar':
+			dateFormats = ['dddd, d MMMM, yyyy', 'dd MMMM, yyyy', 'dd/MM/yyyy', 'MMMM, yyyy', 'd MMMM'];
+			dayPeriods = ['ص', 'م'];
+			months[0] = months[1] = ['يناير', 'فبراير', 'مارس', 'أبريل', 'مايو', 'يونيو', 'يوليو', 'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر'];
+			days = [
+				['الأحد', 'الإثنين', 'الثلاثاء', 'الأربعاء', 'الخميس', 'الجمعة', 'السبت'],
```

Comment:
It looks like the translation for Sunday has been moved to the last position, and Saturday is now first in the array. Is this expected?

Reply:
Looks like this is just a bug in GitHub's web UI -- in the actual source code it's first.